### ISSUE TITLE
Syntax highlighting is broken with array syntax

### ISSUE THREAD

**Issue by Yanpas** [redhat-developer/vscode-java#728]
[provide a description of the issue]

##### Environment
- Operating System: linux
- JDK version: 11
- Visual Studio Code version: 1.29.1
- Java extension version: 0.34

##### Steps To Reproduce
Snippet:
```java
        for (int i = 0; i==0;){
            
        }
        String[] aaa = new String[] {
            String.class.toString(),
            "aaaa",
            "bbb",
        };
        for (int i = 0; i==0;){

        }
```
![image](https://user-images.githubusercontent.com/7345761/49146556-71902380-f313-11e8-910e-a7e642d590fa.png)

**Comment by fbricon** [comment]
Looks like this is caused by this issue upstream: https://github.com/atom/language-java/issues/172

**Comment by fbricon** [comment]
@snjeza please keep an eye on this one. You can provide feedback on the proposed fix upstream, if it can help getting things moved forward.

**Comment by Vigilans** [comment]
Also reproduced when initialized with new operator:
![image](https://user-images.githubusercontent.com/20227484/66463577-ee516980-eaaf-11e9-871e-7cf4e9cbc57b.png)

After new keyword gets removed:
![image](https://user-images.githubusercontent.com/20227484/66463662-1e990800-eab0-11e9-827d-04c0edeb70e6.png)

**Comment by nkibbey** [comment]
Found that some array syntax still breaks some highlighting 

using 
vscode 1.43.2
java extension 0.59.1
java 8

![poorHighlighting](https://user-images.githubusercontent.com/16986162/78788319-a38aae80-7979-11ea-8b4c-e646f6359d57.PNG)

**Comment by Eskibear** [comment]
@nkibbey It has been fixed in upstream, and has been adopted in vscode-insiders. I just tried in latest vscode-insiders, and it worked. So it's expected to be shipped in the next stable release of vscode.

**Comment by fbricon** [comment]
Works in stable vscode 1.44.0

**Comment by Yanpas** [comment]
@fbricon Vscode 1.44.2, the following snippet doesn't work:
```java
class A {
public static void name() {
	
	for (int i = 0; i==0;){
		
	}
	String[] aaa = new String[] {
		String.class.toString(),
		"aaaa",
		"bbb",
	};
	for (int i = 0; i==0;){
		
	}
}
}
```

Maybe it's https://github.com/microsoft/vscode/blob/master/extensions/java/syntaxes/java.tmLanguage.json problem

**Comment by fbricon** [comment]
@Yanpas sorry I did update my previous comment. Your issue is fixed in upcoming 0.61.0, with semantic highlighting turned on:

<img width="399" alt="Screen Shot 2020-04-27 at 11 48 29 PM" src="https://user-images.githubusercontent.com/148698/80424251-9fe7a900-88e1-11ea-86ff-09e49da54859.png">
